src/OneExplorer/ConfigObject.ts

```diff
@@ -215,37 +216,49 @@ export class ConfigObj {
    * a parsed config object
    */
   obj: {
-    baseModels: vscode.Uri[],
-    derivedModels: vscode.Uri[],
+    baseModels: Artifact[],
+    derivedModels: Artifact[],
   };
 
-  private constructor(uri: vscode.Uri, rawObj: object) {
-    this.uri = uri;
-    this.rawObj = rawObj;
-    this.obj = {
-      baseModels: ConfigObj.parseBaseModels(uri, rawObj),
-      derivedModels: ConfigObj.parseDerivedModels(uri, rawObj)
-    };
+  get getBaseModels() {
+    return this.obj.baseModels;
+  }
+
+  get getDerivedModels() {
+    return this.obj.baseModels;
   }
 
   /**
-   * A simple parse function with directly `ConfigObj.obj` fields after parsing
-   * without keeping an ConfigObj instance.
-   *
-   * @param uri
-   * @returns `{baseModels, derivedModels}` or
-   *          `null` if failed to open/parse into ini object
+   * Returns only the baseModels which exists in file system
+   */
+  get getBaseModelsExists() {
+    return this.obj.baseModels.filter(artifact => RealPath.exists(artifact.path!));
   }
+
+  /**
+   * Returns only the derivedModels which exists in file system
    */
-  public static parse(uri: vscode.Uri): {baseModels: vscode.Uri[],
-                                         derivedModels: vscode.Uri[]}|null {
-    const cfgObj = this.createConfigObj(uri);
+  get getDerivedModelsExists() {
+    return this.obj.derivedModels.filter(artifact => RealPath.exists(artifact.path!));
+  }
 
-    if (!cfgObj) {
-      return null;
-    }
+  /**
+   * Return true if the `baseModelPath` is included in `baseModels`
+   */
+  public isChildOf(baseModelPath: string): boolean {
+    const found = this.obj.baseModels.map(artifact => artifact.path!)
+                      .find(path => RealPath.areEqual(baseModelPath, path!));
```

Comment:
```suggestion
                      .find(path => path && RealPath.areEqual(baseModelPath, path));
```

(optional)
If the path is always defined, ignore my comment.

Reply:
Yes, the path is always defined by the point.

Reply:
Then, how about using `path: string;` instead of `path?: string;`?

from [src/OneExplorer/ConfigObject.ts](https://github.com/Samsung/ONE-vscode/pull/1016/files#diff-59c113e60e8574721c44c2b6f9f619191c57a5b338e920de950af5ca9db2f728L42)

Reply:
It's not possible for the moment, as the interface 'Artifact' is used when it asks 'ArtifactLocator' to find the path...

So 'ArtifactLocator' is given an Artifact{type, ext} and it returns Artifact{type, ext, path}.

I think that it's not a good design to use the interface for both query and return.. I think I need to rewrite to distinguish them. 

I will remove artifact from the 'artifact locator' query, and do as what you said: use `path: string;` instead of `path?: string;`

Reply:
I committed the change. 😄  https://github.com/Samsung/ONE-vscode/pull/1016#discussion_r919723520